tests/Integration/TracerTest.php

```diff
@@ -234,6 +234,72 @@ function ($execute) {
         $this->assertSame('custom-resource', $traces[0][0]['resource']);
     }
 
+    public function testEnvironmentIsAddedToASpan()
+    {
+        $this->putEnvAndReloadConfig(['DD_ENV=my-env']);
+        $traces = $this->isolateTracer(function (Tracer $tracer) {
+            $scope = $tracer->startActiveSpan('custom.root');
+            $scope->close();
+        });
+
+        $this->assertSame('my-env', $traces[0][0]['meta']['env']);
+    }
+
+    public function testEnvironmentIsOptional()
+    {
+        $this->putEnvAndReloadConfig(['DD_ENV']);
+        $traces = $this->isolateTracer(function (Tracer $tracer) {
+            $scope = $tracer->startActiveSpan('custom.root');
+            $scope->close();
+        });
+
+        $this->assertTrue(empty($traces[0][0]['meta']['env']));
+    }
+
+    public function testDDEnvHasPrecendenceOverGlobalTags()
```

Comment:
Note, this would succeed in any case until the `DD_TAG` PR #920 is merged. After #920 is merged then it succeed only if the order it correct.